sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosAuthorizationTokenResolver.java

```diff
@@ -1,33 +1,33 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-package com.azure.cosmos.implementation;
+package com.azure.cosmos.models;
+
+import com.azure.cosmos.util.Beta;
 
 import java.util.Map;
 
 /**
  * This interface is for client side implementation, which can be used for initializing
- * AsyncDocumentClient without passing master key, resource token and permission feed.<br>
+ * CosmosAsyncClient without passing master key, resource token and permission feed.<br>
  * <br>
  * Each time the SDK create request for CosmosDB, authorization token is generated based on that
- * request at client side which enables creation of one AsyncDocumentClient per application shared across various users
+ * request at client side which enables creation of one CosmosAsyncClient per application shared across various users
  * with different resource permissions.
  */
-@FunctionalInterface
+@Beta(value = Beta.SinceVersion.V4_24_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
```

Comment:
who is the customer for this new public API?

as the future path of authentication for many customers is AAD why can't we direct the customer to use AAD?